src/SourceBuild/content/eng/finish-source-only.proj

```diff
@@ -30,34 +30,15 @@
                           OutputPath="$(ArtifactsLogDir)" />
   </Target>
 
-  <UsingTask TaskName="IndexItemGroup" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll" >
-    <ParameterGroup>
-      <Items Required="true" ParameterType="Microsoft.Build.Framework.ITaskItem[]"/>
-      <Index Required="true" ParameterType="System.Int32"/>
-      <Item Output="true" ParameterType="Microsoft.Build.Framework.ITaskItem"/>
-    </ParameterGroup>
-    <Task>
-      <Code Type="Fragment" Language="cs">
-        <![CDATA[Item = Items[ Index ];]]>
-      </Code>
-    </Task>
-  </UsingTask>
-
   <Target Name="DetermineSourceBuiltSdkNonStableVersion"
           DependsOnTargets="DetermineSourceBuiltSdkVersion">
     <ItemGroup>
       <SdkVersionFileItem Include="$(DotNetSdkExtractDir)/sdk/**/.version" />
     </ItemGroup>
 
-    <ReadLinesFromFile File="%(SdkVersionFileItem.Identity)">
-      <Output
-          TaskParameter="Lines"
-          ItemName="VersionFileLines"/>
-    </ReadLinesFromFile>
-
-    <IndexItemGroup Items="@(VersionFileLines)" Index="3">
-      <Output PropertyName="SourceBuiltSdkNonStableVersion" TaskParameter="Item"/>
-    </IndexItemGroup>
+    <PropertyGroup>
+      <SourceBuiltSdkNonStableVersion>$([System.Text.RegularExpressions.Regex]::Split('$([System.IO.File]::ReadAllText('%(SdkVersionFileItem.Identity)'))', '\r\n|\r|\n')[3])</SourceBuiltSdkNonStableVersion>
```

Comment:
Can `ReadAllLines` be used instead to simplify the logic?

Reply:
I could not get `ReadAllLines` to work in MSBuild: `error MSB4185: The function "ReadAllLines" on type "System.IO.File" is not available for execution as an MSBuild property function`

There is a built-in task as an alternative - `ReadLinesFromFile`. But, then the code gets a bit larger.

Reply:
Yeah not sure why we allow `ReadAllText` but not `ReadAllLines`; I'd be willing to expand that if you care to file it. But this seems ok (or using `string.Split()`).